tests/e2e/test_read.py

```diff
@@ -39,29 +44,100 @@ def container_client(account_url):
     container.delete_container()
 
 
-@pytest.fixture
-def blob_client(container_client, sample_data):
+@pytest.fixture(scope="module")
+def small_uploaded_blob(account_url, container_client):
     blob_name = random_resource_name()
+    data = sample_data(20)
     blob_client = container_client.get_blob_client(blob=blob_name)
-    blob_client.upload_blob(sample_data)
-    return blob_client
+    blob_client.upload_blob(data)
+    url = f"{account_url}/{container_client.container_name}/{blob_name}"
 
+    return Blob(data=data, url=url)
 
-@pytest.fixture
-def blob_url(account_url, container_client, blob_client):
-    return f"{account_url}/{container_client.container_name}/{blob_client.blob_name}"
+
+@pytest.fixture(scope="module")
+def large_uploaded_blob(account_url, container_client):
+    blob_name = random_resource_name()
+    data = sample_data(_PARTITIONED_DOWNLOAD_THRESHOLD * 2)
+    blob_client = container_client.get_blob_client(blob=blob_name)
+    blob_client.upload_blob(data)
+    url = f"{account_url}/{container_client.container_name}/{blob_name}"
+
+    return Blob(data=data, url=url)
+
+
+@pytest.fixture(scope="module")
+def small_with_newlines_uploaded_blob(account_url, container_client):
+    blob_name = random_resource_name()
+    data = sample_data(20, True)
+    blob_client = container_client.get_blob_client(blob=blob_name)
+    blob_client.upload_blob(data)
+    url = f"{account_url}/{container_client.container_name}/{blob_name}"
+
+    return Blob(data=data, url=url)
 
 
 @pytest.fixture
-def blob_io(blob_url):
-    yield BlobIO(blob_url=blob_url, mode="rb")
+def uploaded_blob(request):
+    if request.param == "small_blob":
+        return request.getfixturevalue("small_uploaded_blob")
+    elif request.param == "large_blob":
+        return request.getfixturevalue("large_uploaded_blob")
+    elif request.param == "small_with_newlines_blob":
+        return request.getfixturevalue("small_with_newlines_uploaded_blob")
 
 
 def random_resource_name(name_length=8):
-    return "".join(random.choices(string.ascii_lowercase + string.digits, k=name_length))
+    return "".join(
+        random.choices(string.ascii_lowercase + string.digits, k=name_length)
+    )
+
+
+def sample_data(data_length=20, newline=False):
```

Comment:
For the `sample_data` function, it may make sense to keep it as is and then we can delegate it for random data. Mainly for generating random newline data we will probably want more granularity such as how long we want each line and how many newlines we want. We can probably just create a new function called `sample_data_with_newlines()` and it can accept a `line_length` and `num_lines` and from there we can call `sample_data` by the number of lines and join with newlines so that we can guarantee that we are getting newlines in our content.